packages/chat-component/src/main.ts

```diff
@@ -377,96 +368,345 @@ export class ChatComponent extends LitElement {
-            top: 3,
-            suggest_followup_questions: false,
+    const response = await fetch(`${globalConfig.API_URL}${type}`, {
+      method: options.method,
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({
+        // must enable history persistence
+        history: [
+          {
+            user: this.currentQuestion,
           },
-        }),
-        headers: {
-          'Content-Type': 'application/json',
-        },
-      })
-        .then((response) => {
-          if (!response.ok) {
-            this.handleAPIError();
-            throw new Error(response.statusText);
-          }
-          const data = response.json();
-          return data;
-        })
-        .then((data) => {
-          // Add the response to the chat messages
-          this.addMessage(data.answer, false);
-          this.isDisabled = false;
-          this.isAwaitingResponse = false;
-        });
-      // Enable the input field and submit button again
-    } catch (error) {
-      console.error('API Response Exception. Error:', error);
+        ],
+        approach: options.approach,
+        overrides: options.overrides,
+        stream: this.isStreaming,
+      }),
+    });
+
+    if (this.isStreaming) {
+      return response;
     }
+
+    const parsedResponse: BotResponse = await response.json();
+    if (response.status > 299 || !response.ok) {
+      this.handleAPIError();
+      throw new Error(response.statusText) || 'API Response Error';
+    }
+    return parsedResponse;
   }
 
-  // Add a message to the chat, when the user or the API sends a message
-  addMessage(message: string, isUserMessage: boolean): void {
-    const citations: Citation[] = [];
-    const followingSteps: string[] = [];
-    const followupQuestions: string[] = [];
-    // Check if message is a bot message to process citations and follow-up questions
-    if (!isUserMessage) {
-      const processedText = processText(message, [citations, followingSteps, followupQuestions]);
-      message = processedText.replacedText;
-      // Push all lists coming from processText to the corresponding arrays
-      citations.push(...(processedText.arrays[0] as unknown as Citation[]));
-      followingSteps.push(...(processedText.arrays[1] as string[]));
-      followupQuestions.push(...(processedText.arrays[2] as string[]));
+  readStream = async function* <T>(response: Response): AsyncGenerator<T, void> {
+    const reader: any = response.body
+      ?.pipeThrough(new TextDecoderStream())
+      .pipeThrough(new EventSourceParserStream())
+      .getReader();
+
+    if (!reader) {
+      throw new Error('No response body or body is not readable');
     }
-    // Get the timestamp for the message
-    const timestamp = this.getTimestamp();
-    // Add the message to the chat messages array
+
+    for (;;) {
+      const { done, value } = await reader.read();
+      if (done || value?.data === 'Stream closed') {
```

Comment:
Did it ever return 'Stream closed'? I could never validate that chunk is sent.

Reply:
never managed to receive that marker.